### DISCUSSION TITLE
Windows gdiplus backend

### DISCUSSION THREAD

**Discussion by amoldeshpande** [alandefreitas/matplotplusplus#430] [Ideas]
Hello,

I was looking for a plotting library to use from C++, in a desktop Windows application and found matplotplusplus very interesting. 

I wrote the beginnings of a GDI+ based backend before I got to the point where I realized the core requires quite a bit more work to make backends first-class replacements to gnuplot commands.

The code isn't in the state I'd want for a pull request (specifically the hacky draw_lines function I added to axes.cpp)  but if anyone is interested in carrying it forward, you can find it here:

https://github.com/amoldeshpande/matplotplusplus

There's a simple example Win32 GUI application you can use to test various plots. 

I chose GDI+ over DirectX or OpenGL because it's quite simple to use and powerful enough for most charting needs I can think of.

**Comment by alandefreitas** [comment]
Yes. I think any serious backend would need some adaptations in the backend API in Matplot++. We're open to that, of course. It would be great to have a backend better than Gnuplot because Gnuplot doesn't scale well. But I guess GDI+ would be limited to Windows. There are still some alternatives at many levels, though. Even higher levels than OpenGL, like SDL, cairo, etc...
